gen_topo.sh

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env sh
+#PBS -q normal
+#PBS -l walltime=4:00:00,mem=4GB
+#PBS -l wd
+#PBS -lstorage=gdata/hh5+gdata/ik11
+
+# Ensure build.sh is executable
+if [ ! -x "./build.sh" ]; then
+  chmod +x ./build.sh
+fi
+
+# Build domain-tools
+./build.sh
+
+module use /g/data/hh5/public/modules
+module load conda/analysis3
+module load nco
+
+set -x
+set -e
+
+# Using the environment variables passed via -v
+INPUT_HGRID=$INPUT_HGRID
+INPUT_VGRID=$INPUT_VGRID
+INPUT_GBCO=$INPUT_GBCO
+
+# Copy and link input files
+cp -L --preserve=timestamps "$INPUT_HGRID" ./ocean_hgrid.nc
+cp -L --preserve=timestamps "$INPUT_VGRID" ./ocean_vgrid.nc
+ln -sf "$INPUT_GBCO" ./GEBCO_2024.nc
+
+# Convert double precision vgrid to single
+./domain-tools/bin/float_vgrid --vgrid ocean_vgrid.nc --vgrid_type mom6
+
+# Interpolate topography on horizontal grid:
+./domain-tools/bin/topogtools gen_topo -i GEBCO_2024.nc -o topog_new.nc --hgrid ocean_hgrid.nc --tripolar --longitude-offset -100 
+
+# Cut off T cells of size less than 6kms (6000 m)
+./domain-tools/bin/topogtools min_dy -i topog_new.nc -o topog_new_min_dy.nc --cutoff 6000 --hgrid ocean_hgrid.nc
+
+# Fill cells that have a sea area fraction smaller than 0.5:
+./domain-tools/bin/topogtools fill_fraction -i topog_new_min_dy.nc -o topog_new_fillfraction.nc  --fraction 0.5
+
+# edit_topo.py
+python ./topogtools/editTopo.py --overwrite --nogui --apply edit_topog_new_fillfraction.txt --output topog_new_fillfraction_edited.nc topog_new_fillfraction.nc
+
+# Remove seas:
+./domain-tools/bin/topogtools deseas -i topog_new_fillfraction_edited.nc -o topog_new_fillfraction_edited_deseas.nc --grid_type C
+
+# Set maximum/minimum depth
+./domain-tools/bin/topogtools min_max_depth -i topog_new_fillfraction_edited_deseas.nc -o topog_new_fillfraction_edited_deseas_mindepth.nc --level 4 --vgrid ocean_vgrid.nc --vgrid_type mom6
+
+# Create land/sea mask
+cp topog_new_fillfraction_edited_deseas_mindepth.nc topog.nc
+./domain-tools/bin/topogtools mask -i topog.nc -o ocean_mask.nc
+
+# Calculate MD5 checksum for topog.nc
+MD5SUM_topog=$(md5sum topog.nc | awk '{print $1}')
+
+# Add MD5 checksum as a global attribute to topog.nc
+ncatted -O -h -a md5_checksum,global,a,c,"$MD5SUM_topog" topog.nc
```

Comment:
```suggestion
ncatted -O -h -a md5_checksum,global,a,c,"$MD5SUM_topog" ocean_mask.nc
```

As `topog.nc` is an input to generate `ocean_mask.nc`, we need to be able to confirm the version of `topog.nc` used as the input. So the md5sum of topog.nc goes in the attributes of the ocean_mask file. I think the suggestion does this ?

Reply:
This suggestion also removes md5sum of `topog.nc`?